modules/p2/ding_list/theme/theme.inc

```diff
@@ -12,6 +12,73 @@ function theme_ding_list_message($variables) {
   return '<div class="ding-list-message">' . $variables['message'] . '</div>';
 }
 
+/**
+ * Preprocesses variables for ding_list_element.
+ */
+function template_preprocess_ding_list_element(&$variables) {
+  $element = $variables['element'];
+  $variables['classes_array'][] = 'ding-list-element--type-' . drupal_html_class($element->type);
+  $variables['classes_array'][] = 'ding-list-element--display-' . drupal_html_class($variables['display']);
+  $variables['classes_array'][] = 'clearfix';
+  $variables['attributes_array']['data-element-id'] = $element->id;
+
+  switch ($element->type) {
+    case 'ting_object':
+      $view_mode_matrix = array(
+        'view list' => 'ting_reference_preview',
+      );
+      $ting_object = ting_object_load(ding_list_convert_cql($element->value));
+      if ($ting_object) {
+        $variables['content']['ting_object'] = ting_object_view(
+          $ting_object,
+          isset($view_mode_matrix[$variables['display']]) ? $view_mode_matrix[$variables['display']] : 'default'
+        );
+      }
+      else {
+        $variables['content']['value'] = array(
+          '#theme' => 'html_tag',
+          '#tag' => 'div',
+          '#value' => t('Missing item'),
+          '#attributes' => array(
+            'class' => array('ding-list-element__title'),
+          ),
+        );
+        drupal_set_message(t('Unable to retrieve some of the items on this list. The list may appear incomplete.'), 'status', FASE);
```

Comment:
`FASE` should should probably be `FALSE`. Status could also be `error` or at lease `warning`?

Do you think it is important to render the item, and not log to watchdog?

Reply:
I've kept the missing item, so the user might have a better idea of what is missing.

Added a watchdog with severity of `error`, and changed the dsm to `warning`. I haven't used the error for the user, because this is handling an error, and the user wont be able to do anythign different.